pkg/sql/logictest/testdata/logic_test/alter_table

```diff
@@ -403,12 +405,16 @@ CREATE TABLE o (gf INT REFERENCES t (g), h INT, i INT, INDEX oi (i), INDEX oh (h
 statement error "t_g_key" is referenced by foreign key from table "o"
 ALTER TABLE t DROP COLUMN g
 
-statement ok
+statement notice NOTICE: dropping index "t_g_key" which depends on column "g"
 ALTER TABLE t DROP COLUMN g CASCADE
 
 # Dropping columns that are indexed or stored in indexes drops those indexes
-# too.
-statement ok
+# too. Excuse the nasty regex, the legacy schema changer emits additional
+# NOTICEs about GCing. We're not testing for that, so we just "eat" them.
+# Example: "NOTICE: the data for dropped indexes is reclaimed
+# asynchronously\nHINT: The reclamation delay can be customized in the zone
+# configuration for the table."
+statement notice (NOTICE: dropping index "(oh|oih)" which depends on column "h"(\nNOTICE: .+\nHINT: .+\n)?\n?){2}
```

Comment:
The regex here is indeed a bit nasty. Can we just do something like "NOTICE: dropping index "(oh|oih)" which depends on column "h".*"

Reply:
I agree that it's pretty nasty! I didn't want to `.*` it because I'm a bit worried of "slurping up" too much. Though, `.` doesn't match `\n`'s by default in go. So maybe that's paranoia? WDYT?

Reply:
i'm OK with the test having a tricky regex, especially since you have a good comment for it.